### ISSUE TITLE
Doc jobs don't use custom rosdep URLs

### ISSUE THREAD

**Issue by ayrton04** [ros-infrastructure/ros_buildfarm#409]
We are having issues resolving dependencies in our doc jobs. We have a custom rodistro setup, and so I decided to compare the issue with some dev jobs, which are able to resolve dependencies just fine. Below is some sample output (server URLs redacted):

**Dev job:**
```
06:11:12   ENV ROSDISTRO_INDEX_URL http://our-rosdistro-server/path/to/master/index.yaml
06:11:12  
06:11:12   RUN rosdep init
06:11:12   RUN (cd /etc/ros/rosdep/sources.list.d/ && wget --no-verbose -O $(basename http://our-rosd-stro-server/path/to/master/rosdep/sources.list.d/20-default.list )  http://our-rosd-stro-server/path/to/master/rosdep/sources.list.d/20-default.list )
(several steps later, we call rosdep update)
```

**Doc job:**
```
05:58:22 Step 19 : ENV ROSDISTRO_INDEX_URL http://our-rosdistro-server/path/to/master/index.yaml
05:58:22  ---> Running in 3077aad905bf
05:58:22  ---> 3396b2975a6c
05:58:22 Removing intermediate container 3077aad905bf
05:58:22 Step 20 : RUN rosdep init
05:58:22  ---> Running in b7a11d655a35
05:58:23 Wrote /etc/ros/rosdep/sources.list.d/20-default.list
(several steps later, we call rosdep update)
```

Looking at their templates, the [devel job](https://github.com/ros-infrastructure/ros_buildfarm/blob/2d14e80509a9055ff55ed22a6ccda18d8533ce5f/ros_buildfarm/templates/devel/devel_create_tasks.Dockerfile.em#L54) uses a [snippet](https://github.com/ros-infrastructure/ros_buildfarm/blob/2d14e80509a9055ff55ed22a6ccda18d8533ce5f/ros_buildfarm/templates/snippet/rosdep_init.Dockerfile.em) for `rosdep init`, whereas the [doc job](https://github.com/ros-infrastructure/ros_buildfarm/blob/2d14e80509a9055ff55ed22a6ccda18d8533ce5f/ros_buildfarm/templates/doc/doc_create_task.Dockerfile.em#L48) just calls it.

Are they intentionally different? If not, I can submit a PR.

**Comment by tfoote** [comment]
That looks like an oversight. A PR would be appreciated.

**Comment by ayrton04** [comment]
Will do!

**Comment by dirk-thomas** [comment]
Implemented in #410.
